waspc/src/Wasp/Project/Deployment.hs

```diff
@@ -27,18 +28,22 @@ loadUserDockerfileContents waspDir = do
 -- | This will run our TS deploy project by passing all args from the Wasp CLI straight through.
 -- The TS project is compiled to JS in CI and included in the data dir for the release archive.
 -- If the project was not yet built locally (i.e. after they just installed a Wasp version), we do so.
-deploy :: FilePath -> Path' Abs (Dir WaspProjectDir) -> [String] -> IO ()
+deploy :: FilePath -> Path' Abs (Dir WaspProjectDir) -> [String] -> IO (Either String ())
 deploy waspExe waspDir cmdArgs = do
   waspDataDir <- Data.getAbsDataDirPath
   let deployDir = waspDataDir </> [reldir|packages/deploy|]
   let nodeModulesDirExists = doesDirectoryExist . toFilePath $ deployDir </> [reldir|node_modules|]
   unlessM nodeModulesDirExists $
-    runCommandAndPrintOutput $ runNodeCommandAsJob deployDir "npm" ["install"] J.Server
+    void $ runCommandAndPrintOutput $ runNodeCommandAsJob deployDir "npm" ["install"] J.Server
   let deployScriptArgs = ["dist/index.js"] ++ cmdArgs ++ ["--wasp-exe", waspExe, "--wasp-project-dir", toFilePath waspDir]
   -- NOTE: Here we are lying by saying we are running in the J.Server context.
   -- TODO: Consider adding a new context for these types of things, like J.Other or J.External.
   runCommandAndPrintOutput $ runNodeCommandAsJob deployDir "node" deployScriptArgs J.Server
   where
+    runCommandAndPrintOutput :: J.Job -> IO (Either String ())
     runCommandAndPrintOutput job = do
       chan <- newChan
-      void $ concurrently (printJobMsgsUntilExitReceived chan) (job chan)
+      (_, exitCode) <- concurrently (printJobMsgsUntilExitReceived chan) (job chan)
+      case exitCode of
+        ExitSuccess -> return $ Right ()
+        ExitFailure code -> return $ Left $ "Deploy command failed with exit code: " ++ show code
```

Comment:
Haha, I am on the fence on this one, only because the existing way is pretty common in the code base. So I may stick with that as is, but I like the thinking here :D Thanks!